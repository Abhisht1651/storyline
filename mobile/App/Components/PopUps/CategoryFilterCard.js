// @flow

import React, { useEffect } from 'react'
import { View, Text, Image, TouchableOpacity, SafeAreaView, FlatList } from 'react-native'
import Images from '../../Themes/Images'
import styles from '../Styles/PopUpStyles/CategoryFilterCardStyle'

import useHooks from './CategoryFilterHooks'

type Props = {
  getSelectedOptions: (arr: Array<string>) => void
}

const CategoryFilterCard = (props: Props) => {
  const { getSelectedOptions } = props
  const {
    numberOfItemsSelected,
    checkedWorld,
    checkedNational,
    checkedScience,
    checkedSports,
    checkedTechnology,
    checkedEntertainment,
    checkedOthers,
    optionProps,
    clickHandler1,
    clickHandler2,
    clickHandler3,
    clickHandler4,
    clickHandler5,
    clickHandler6,
    clickHandler7,
    addPropertyHandler,
    removePropertyHandler,
    clearAll
  } = useHooks()

  const selectedOptions = []

  useEffect(() => {
    if (checkedWorld) {
      selectedOptions.push('World')
    }
    if (checkedNational) {
      selectedOptions.push('National')
    }
    if (checkedSports) {
      selectedOptions.push('Sports')
    }
    if (checkedScience) {
      selectedOptions.push('Science')
    }
    if (checkedEntertainment) {
      selectedOptions.push('Entertainment')
    }
    if (checkedTechnology) {
      selectedOptions.push('Technology')
    }
    if (checkedOthers) {
      selectedOptions.push('Others')
    }
    getSelectedOptions(selectedOptions)
  }, [
    checkedWorld,
    checkedNational,
    checkedSports,
    checkedScience,
    checkedEntertainment,
    checkedTechnology,
    checkedOthers
  ])

  const data = [
    {
      id: 1,
      title: 'World',
      image: Images.world,
      stateProp: checkedWorld,
      functionCall: clickHandler1
    },
    {
      id: 2,
      title: 'National',
      image: Images.national,
      stateProp: checkedNational,
      functionCall: clickHandler2
    },
    {
      id: 3,
      title: 'Sports',
      image: Images.sports,
      stateProp: checkedSports,
      functionCall: clickHandler3
    },
    {
      id: 4,
      title: 'Science',
      image: Images.science,
      stateProp: checkedScience,
      functionCall: clickHandler4
    },
    {
      id: 5,
      title: 'Entertainment',
      image: Images.entertainment,
      stateProp: checkedEntertainment,
      functionCall: clickHandler5
    },
    {
      id: 6,
      title: 'Technology',
      image: Images.technology,
      stateProp: checkedTechnology,
      functionCall: clickHandler6
    },
    {
      id: 7,
      title: 'Others',
      image: Images.others,
      stateProp: checkedOthers,
      functionCall: clickHandler7
    }
  ]

  return (
    <SafeAreaView style={styles.popUpCard}>
      <View style={styles.categoryFilter}>
        <Text style={styles.categoryFilterText}>Category Filter</Text>
      </View>
      <FlatList
        nestedScrollEnabled
        onScroll={addPropertyHandler}
        onMomentumScrollEnd={removePropertyHandler}
        contentContainerStyle={styles.flatList}
        data={data}
        renderItem={({ item }) => (
          <View style={styles.cardItems}>
            <View style={styles.iconContainer}>
              <Image source={item.image} style={styles.iconStyles} />
            </View>
            <View style={styles.itemText}>
              <Text style={styles.text}>{item.title}</Text>
            </View>
            <View style={styles.checkboxContainer}>
              <TouchableOpacity style={styles.checkBox} onPress={item.functionCall}>
                {item.stateProp && (
                  <View style={styles.checkedBox}>
                    <Image source={Images.checkmark} style={styles.checkmark} />
                  </View>
                )}
              </TouchableOpacity>
            </View>
          </View>
        )}
        keyExtractor={(item) => item.id.toString()}
      />
      <View style={[styles.options, optionProps]}>
        <View>
          {numberOfItemsSelected > 0 && (
            <TouchableOpacity activeOpacity={0.6}>
              <Text style={styles.selectedText}>{numberOfItemsSelected} Selected</Text>
            </TouchableOpacity>
          )}
        </View>

        <TouchableOpacity activeOpacity={0.6} onPress={clearAll}>
          <Text style={styles.clearSelectionText}>Clear Selection</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.footer}>
        <View style={styles.bottomLine} />
      </View>
    </SafeAreaView>
  )
}

export default CategoryFilterCard
