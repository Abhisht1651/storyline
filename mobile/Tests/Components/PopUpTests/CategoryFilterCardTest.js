import { shallow } from 'enzyme'
import React from 'react'
import { renderHook, act } from '@testing-library/react-hooks'
import CategoryFilterCard from '../../../App/Components/PopUps/CategoryFilterCard'

import useHooks from '../../../App/Components/PopUps/CategoryFilterHooks'

jest.useFakeTimers()

describe('CategoryFilterCard', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<CategoryFilterCard />)
      expect(component).toMatchSnapshot()
    })
  })
})

describe('CategoryFilter Popup', () => {
  it('item selected should increase /decrese if any item is selected', () => {
    const { result } = renderHook(useHooks)
    const val1 = result.current.numberOfItemsSelected
    const val2 = result.current.checkedWorld
    if (val2 === true) {
      act(() => {
        result.current.clickHandler1()
      })
      expect(result.current.numberOfItemsSelected).toBe(val1 - 1)
    } else {
      act(() => {
        result.current.clickHandler1()
      })
      expect(result.current.numberOfItemsSelected).toBe(val1 + 1)
    }
  })

  it('should clear all items ', () => {
    const { result } = renderHook(useHooks)
    act(() => {
      result.current.clearAll()
    })
    expect(result.current.numberOfItemsSelected).toBe(0)
  })
})
