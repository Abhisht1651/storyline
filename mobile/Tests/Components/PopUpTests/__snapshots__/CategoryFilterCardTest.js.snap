// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilterCard Rendering should match to snapshot 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 19.5,
      "borderWidth": 1,
      "height": 805,
      "paddingLeft": 0,
      "paddingRight": 0,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#E6ECED",
        "height": 100.625,
        "paddingTop": 24.642857142857142,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2C2A81",
          "fontFamily": "helvetica",
          "fontSize": 23.4,
          "letterSpacing": -0.12,
        }
      }
    >
      Category Filter
    </Text>
  </View>
  <FlatList
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    data={
      Array [
        Object {
          "functionCall": [Function],
          "id": 1,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "World",
        },
        Object {
          "functionCall": [Function],
          "id": 2,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "National",
        },
        Object {
          "functionCall": [Function],
          "id": 3,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "Sports",
        },
        Object {
          "functionCall": [Function],
          "id": 4,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "Science",
        },
        Object {
          "functionCall": [Function],
          "id": 5,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "Entertainment",
        },
        Object {
          "functionCall": [Function],
          "id": 6,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "Technology",
        },
        Object {
          "functionCall": [Function],
          "id": 7,
          "image": Object {
            "default": "",
          },
          "stateProp": false,
          "title": "Others",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    nestedScrollEnabled={true}
    numColumns={1}
    onEndReachedThreshold={2}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "borderColor": "#E6ECED",
          "flexDirection": "row",
          "height": 49.285714285714285,
          "justifyContent": "space-between",
          "width": "100%",
        },
        Object {},
      ]
    }
  >
    <View />
    <ForwardRef
      activeOpacity={0.6}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 23.4,
            "letterSpacing": -0.09,
            "paddingRight": 48,
          }
        }
      >
        Clear Selection
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 48.05357142857142,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E6ECED",
          "borderRadius": 3,
          "height": 6.571428571428571,
          "width": 90,
        }
      }
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
